src/zepben/edith/__init__.py

```diff
@@ -3,56 +3,94 @@
 #  This Source Code Form is subject to the terms of the Mozilla Public
 #  License, v. 2.0. If a copy of the MPL was not distributed with this
 #  file, You can obtain one at https://mozilla.org/MPL/2.0/.
-
+import itertools
+import random
 from asyncio import get_event_loop
-from typing import List
 from zepben.evolve import *
+from zepben.protobuf.nc.nc_requests_pb2 import INCLUDE_ENERGIZED_LV_FEEDERS
 
 
-def do_nothing_allocator(_):
-    pass
+def do_nothing_allocator(_1, _2):
+    return 0
 
 
-def distribution_transformer_proportional_allocator_creator(proportion: int, edith_customers: List[str]) -> Callable[[LvFeeder], None]:
+def usage_point_proportional_allocator(
+        proportion: int,
+        edith_customers: List[str]
+) -> Callable[[LvFeeder, NameType], int]:
     """
-    Creates an allocator for the synthetic feeder creator that distributes a `proportion` of NMIs from `edith_customers` to an `LvFeeder`
+    Creates an allocator for the synthetic feeder creator that distributes a `proportion` of NMIs from `edith_customers`
+    to an `LvFeeder`
 
     :param proportion: The percentage of Edith customers to distribute to an `LvFeeder`. Must be between 1 and 100
     :param edith_customers: The Edith NMIs to distribute to the `LvFeeder`
-    :return: A function that takes an `LvFeeder` and distributes the `proportion` of NMIs across the EnergyConsumers on the `LvFeeder`.
+    :return: A function that takes an `LvFeeder` and distributes the `proportion` of NMIs across the UsagePoints on the
+    `LvFeeder`.
     """
-    if not 1 < proportion <= 100:
+    if not 1 <= proportion <= 100:
         raise ValueError("Proportion must be between 1 and 100")
 
-    def allocator(lv_feeder: LvFeeder):
-        # Note - given the same LvFeeder and proportion, this function should produce the same result.
-        # if we can assume that the equipment on the LvFeeder is always in the same order, this can
-        # be something simple where you just modify every X EnergyConsumer on the feeder or something (say where X is derived from the number
-        # of EnergyConsumers on the feeder)
-        pass
+    nmi_generator = itertools.cycle(edith_customers)
```

Comment:
Adam requested that we just exhaust the customers rather than try to reach the proportion. Can you change it to stop when the customers are exhausted, or provide a flag that allows you to "cycle" but defaults to not cycle?

Reply:
Done, although this may lead to some LV feeders getting the correct proportion and others getting none, due to how we go through them one-by-one.